File: frontend/src/api/family.ts
import client from './client';

export interface Family {
  family_id: string;
  name: string;
  created_by: string;
  created_at: string;
  members: FamilyMember[];
}

export interface FamilyMember {
  id: string;
  family_id: string;
  user_id: string;
  role: string;
  joined_at: string;
  user_name?: string;
  user_email?: string;
}

export const familyApi = {
  create: async (name: string) => {
    const response = await client.post('/family', { name });
    return response.data;
  },

  getAll: async (): Promise<Family[]> => {
    const response = await client.get('/family');
    return response.data;
  },

  getById: async (familyId: string): Promise<Family> => {
    const response = await client.get(`/family/${familyId}`);
    return response.data;
  },

  inviteMember: async (familyId: string, email: string, role: string) => {
    const response = await client.post(`/family/${familyId}/invite`, { email, role });
    return response.data;
  },

  removeMember: async (familyId: string, memberId: string) => {
    const response = await client.delete(`/family/${familyId}/members/${memberId}`);
    return response.data;
  },
};

File: app/schemas/family.py
from pydantic import BaseModel, EmailStr
from typing import Optional, List
from datetime import datetime
from uuid import UUID
from app.models.family import MemberRole


class FamilyBase(BaseModel):
    name: str


class FamilyCreate(FamilyBase):
    pass


class FamilyResponse(FamilyBase):
    family_id: UUID
    created_by: UUID
    created_at: datetime
    
    class Config:
        from_attributes = True


class FamilyMemberBase(BaseModel):
    role: MemberRole = MemberRole.ADMIN


class FamilyMemberCreate(FamilyMemberBase):
    email: EmailStr


class FamilyMemberInvite(BaseModel):
    email: EmailStr
    role: MemberRole = MemberRole.VIEW_ONLY


class FamilyMemberResponse(FamilyMemberBase):
    id: UUID
    family_id: UUID
    user_id: UUID
    joined_at: datetime
    user_name: Optional[str] = None
    user_email: Optional[str] = None
    
    class Config:
        from_attributes = True


class FamilyWithMembers(FamilyResponse):
    members: List[FamilyMemberResponse] = []
